products/opensuse/main.pm

```diff
@@ -815,6 +819,14 @@ elsif (get_var("SUPPORT_SERVER")) {
         loadtest "support_server/wait";
     }
 }
+elsif (get_var("XEN") || check_var("HOST_HYPERVISOR", "xen")) {
+    load_boot_tests();
+    load_inst_tests();
+    return 1 if get_var('EXIT_AFTER_START_INSTALL');
+    loadtest "virt_autotest/login_console";
+    loadtest "virt_autotest/update_package";
+    loadtest "virt_autotest/reboot_and_wait_up_normal2";
+}
```

Comment:
[DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself). You are calling the same code as already in products/sle/main.pm, please [extract method](http://refactoring.com/catalog/extractMethod.html) to `lib/main_common.pm`

Reply:
Got it, thank you for your great suggestion and comments.